File: fix_port.py
import os
from supabase import create_client

def load_env():
    env = {}
    with open(".env.local") as f:
        for line in f:
            if '=' in line and not line.startswith('#'):
                k, v = line.strip().split('=', 1)
                env[k] = v.strip('"\'')
    return env

env = load_env()
url = env.get("NEXT_PUBLIC_SUPABASE_URL")
key = env.get("SUPABASE_SERVICE_ROLE_KEY")
supabase = create_client(url, key)

res = supabase.table('portaria_dispositivos').update({'porta': 80}).eq('ip', '192.168.1.75').execute()
print(res.data)

File: fix_port.ts
import { createClient } from '@supabase/supabase-js'

const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL!, process.env.SUPABASE_SERVICE_ROLE_KEY!)

async function main() {
  const { data, error } = await supabase.from('portaria_dispositivos').select('id, porta, ip').eq('ip', '192.168.1.75')
  if (data && data.length > 0) {
    const id = data[0].id
    await supabase.from('portaria_dispositivos').update({ porta: 80 }).eq('id', id)
    console.log("Porta atualizada para 80 com sucesso.")
  } else {
    console.log("Catraca não encontrada.")
  }
}
main()
